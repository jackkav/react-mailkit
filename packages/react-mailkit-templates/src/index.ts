export { default as Amario } from './amario';
export { default as Arturia } from './arturia';
export { default as Austin } from './austin';
export { default as BlackFriday } from './blackFriday';
export { default as Card } from './card';
export { default as Christmas } from './christmas';
export { default as HAndm } from './hAndm';
export { default as HappyNewYear } from './happyNewYear';
export { default as OnePage } from './onePage';
export { default as Pizza } from './pizza';
export { default as Proof } from './proof';
export { default as Racoon } from './racoon';
export { default as ReactivationEmail } from './reactivationEmail';
export { default as RealEstate } from './realEstate';
export { default as Recast } from './recast';
export { default as ReceiptEmail } from './receiptEmail';
export { default as ReferralEmail } from './referralEmail';
export { default as TicketShop } from './ticketShop';
export { default as UggRoyale } from './uggRoyale';
export { default as WelcomeEmail } from './welcomeEmail';
export { default as Worldly } from './worldly';

export const templates = [
  'Amario',
  'Arturia',
  'Austin',
  'Black Friday',
  'Card',
  'Christmas',
  'hAndm',
  'Happy New Year',
  'One Page',
  'Pizza',
  'Proof',
  'Racoon',
  'Reactivation Email',
  'Real Estate',
  'Recast',
  'Receipt Email',
  'Referral Email',
  'Ticket Shop',
  'Ugg Royale',
  'Welcome Email',
  'Worldly'
];
