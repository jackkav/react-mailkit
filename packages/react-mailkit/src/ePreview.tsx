import { createElement, FC } from 'react';

export type IPreviewProps = {
	/**
	 * Children
	 */
	children: string;
};

export const EPreview: FC<IPreviewProps> = ({ children }) =>
	createElement('mj-preview', null, children);
