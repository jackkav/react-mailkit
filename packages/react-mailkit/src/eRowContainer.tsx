import React, { ReactElement } from 'react';
import map from 'lodash.map';

import { ESection, ISectionProps } from './eSection';
import { EColumn, IColumnProps } from './eColumn';

export type IRowContainerProps = {
  layout: string;
  sectionProps?: ISectionProps;
  columnProps?: IColumnProps;
  children: ReactElement[];
};

export const RowContainer = (props: IRowContainerProps) => {
  const { layout, sectionProps, columnProps, children } = props;
  if (!layout || layout.length === 0) {
    throw new Error('You must specify a "layout" attribute for SocialMedia');
  }

  const layoutLengths: string[] = layout.split('/');

  if (
    !children ||
    children.length === 0 ||
    layoutLengths.length !== children.length
  ) {
    throw new Error(
      'You must to have the same amount of children as in the layout'
    );
  }

  return (
    <ESection {...sectionProps}>
      {map(children, (child, index) => {
        const columnwidth = `${Number(layoutLengths[index])}0%`;
        return (
          <EColumn key={`col-${index}`} {...columnProps} width={columnwidth}>
            {child}
          </EColumn>
        );
      })}
    </ESection>
  );
};

RowContainer.displayName = 'RowContainer';
