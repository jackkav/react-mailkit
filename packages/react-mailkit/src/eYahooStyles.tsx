import React from 'react';

import { EStyle } from './eStyle';

export type IYahooStylesProps = {
	/**
	 * Height for badge size
	 */
	children: string;
};

export const EYahooStyles = (props: IYahooStylesProps) => {
	const { children } = props;

	if (!children || children.trim().length === 0) {
		throw new Error('You must specify a "children" for Comment');
	}

	return <EStyle>{`@media screen yahoo {${children}}`}</EStyle>;
};
