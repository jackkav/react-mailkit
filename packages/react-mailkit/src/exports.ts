/* EXPORT_COMPONENTS */
export * from './eAll';
export * from './eSelector';
export * from './eHtmlAttributes';
export * from './eHtmlAttribute';
export * from './eClass';
export * from './eHead';
export * from './eInclude';
export * from './eAttributes';
export * from './eBreakpoint';
export * from './eFont';
export * from './ePreview';
export * from './eStyle';
export * from './eTitle';
export * from './eWrapper';
export * from './eTable';
export * from './eSpacer';
export * from './eSocialElement';
export * from './eSocial';
export * from './eRaw';
export * from './eNavbarLink';
export * from './eNavbar';
export * from './eImage';
export * from './eHero';
export * from './eGroup';
export * from './eDivider';
export * from './eCarouselImage';
export * from './eCarousel';
export * from './eAccordionText';
export * from './eAccordionTitle';
export * from './eAccordionElement';
export * from './eAccordion';
export * from './eMail';
export * from './eText';
export * from './eColumn';
export * from './eSection';
export * from './eBody';
export * from './eButton';

/* EXPORT_CUSTOM_COMPONENTS */
export * from './eImageText';
export * from './eQrCode';
export * from './eAppStore';
export * from './eComment';
export * from './eMap';
export * from './eRating';
export * from './eAvatar';
export * from './eUnderlinedTitle';
export * from './eYoutube';
export * from './eYoutubeThumbnail';
export * from './eBarcode';
export * from './eTracking';
export * from './eYahooStyles';
export * from './eSocialMedia';
export * from './eSocialPost';
