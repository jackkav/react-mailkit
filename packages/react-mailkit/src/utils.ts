import { ReactElement } from 'react';
import ReactDOMServer from 'react-dom/server';

const Color = require('color');

export const kebabCase = (string: string) =>
  string.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();

export const boolToString = (name: any, value: any) =>
  value ? name : undefined;

export const toPx = (name: any, value: any) => {
  if (typeof value === 'number') {
    return `${value}px`;
  }

  return value;
};

export const toColor = (name: any, value: any) => {
  // Initialize the color.
  let color = null;

  // Create the actual color based on the given one.
  try {
    color = new Color(value);
  } catch (e) {
    console.warn(e);
  }

  // Return empty string if color does not exist.
  if (!color) {
    return '';
  }

  // Add the alpha for this color if it is not already.
  if (value[0] === '#' && value.length === 9) {
    const alpha = color.alpha().toFixed(2);
    return color.rgb().alpha(alpha).toString();
  }

  // Otherwise return the color as it is.
  return value;
};

/**
 * This function convert react-mailkit to mjml format
 * @param email The react-mailkit component
 * @returns The converted mjml format
 */
export const convertToMjml = (email: ReactElement) =>
  ReactDOMServer.renderToStaticMarkup(email);
