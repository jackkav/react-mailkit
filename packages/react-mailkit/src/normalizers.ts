import { boolToString, toColor, kebabCase, toPx } from './utils';

type IHandlers = {
	[key: string]: any;
};

export const normalizeProps = (props: any) => {
	return Object.keys(props).reduce((acc, curr) => {
		const mjmlProp = kebabCase(curr);

		return {
			...acc,
			[mjmlProp]: handleMjmlProp(mjmlProp, props[curr]),
		};
	}, {});
};

const handleMjmlProp = (name: any, value: any) => {
	if (typeof value === 'undefined' || value === null) {
		return undefined;
	}

	const handler = handlers[name] || ((_name: any, value_: any) => value_);

	return handler(name, value);
};

const handlers: IHandlers = {
	inline: boolToString,
	'full-width': boolToString,
	width: toPx,
	height: toPx,
	'border-radius': toPx,
	'border-width': toPx,
	'background-size': toPx,
	padding: toPx,
	'padding-top': toPx,
	'padding-right': toPx,
	'padding-bottom': toPx,
	'padding-left': toPx,
	'font-size': toPx,
	'letter-spacing': toPx,
	'line-height': toPx,
	'icon-padding': toPx,
	'text-padding': toPx,
	color: toColor,
	'border-color': toColor,
	'background-color': toColor,
	'container-background-color': toColor,
	'inner-background-color': toColor,
	thumbnails: boolToString,
};
