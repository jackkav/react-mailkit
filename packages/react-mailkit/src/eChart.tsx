import { createElement, FC } from 'react';

import { normalizeProps } from './normalizers';

export type IChartProps = {
  children: any;
};

export const EChart: FC<IChartProps> = props => {
  const { children, ...rest } = props;

  return createElement('mj-chart', normalizeProps(rest), children);
};

EChart.displayName = 'EChart';
