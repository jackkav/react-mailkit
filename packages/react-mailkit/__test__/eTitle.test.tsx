import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { ETitle } from '../src/eTitle';

describe('should be correct output', () => {
	it('should not be undefined', () => {
		const { container } = render(<ETitle>Hello Fastmail!</ETitle>);
		expect(container).not.toBeUndefined();
	});

	it('should render ETitle', () => {
		const { container } = render(<ETitle>Hello Fastmail!</ETitle>);
		expect(container.querySelector('mj-title')).toBeInTheDocument();
	});

	it('should have correct content', () => {
		const { container } = render(<ETitle>Hello Fastmail!</ETitle>);
		expect(container.querySelector('mj-title')).toHaveTextContent(
			'Hello Fastmail!',
		);
	});

	it('renders correctly', () => {
		expect(
			renderer.create(<ETitle>Hello Fastmail!</ETitle>).toJSON(),
		).toMatchSnapshot();
	});
});
