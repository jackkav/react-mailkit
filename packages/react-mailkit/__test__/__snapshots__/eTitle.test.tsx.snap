// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be correct output renders correctly 1`] = `
<mj-title>
  Hello Fastmail!
</mj-title>
`;
